server/src/main/java/org/opensearch/cluster/metadata/IndexMetadata.java

```diff
@@ -282,6 +282,16 @@ public Iterator<Setting<?>> settings() {
         Property.IndexScope
     );
 
+    /**
+     * Used to specify SEGMENT replication type as default for all indices. By default, this is false.
+     */
+    public static final Setting<Boolean> DEFAULT_REPLICATION_TYPE_SETTING_SEGMENT = Setting.boolSetting(
+        "opensearch.index.default.replication.type.segment",
```

Comment:
I would rename it to  something along the lines of `indices.replication.strategy` and set the default value to doc rep (preferably using an enum). The user can then set it to `seg_rep` as needed. 
I would also move it under `IndicesService` similar to - https://github.com/opensearch-project/OpenSearch/blob/f40fa82e3667825fd0ba8d4bb17b6959ff7cc4f1/server/src/main/java/org/opensearch/indices/IndicesService.java#L214-L219